### ISSUE TITLE
Does not check nested code blocks (only lints top-level code blocks)

### ISSUE THREAD

**Issue by calebeby** [mdx-js/eslint-mdx#470]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

I am using eslint-plugin-mdx 2.2.0, eslint-mdx 2.2.0

### Link to runnable example

https://stackblitz.com/edit/github-nxeabp?file=testcase.mdx

### Steps to reproduce

Here is my mdx file that I am linting:

````mdx
<main>
  ```js
  im_not_camel_case_oops
  ```
</main>

```js
im_not_camel_case_oops
```
````

ESLint only runs on the top-level code block. The code block inside the `<main>` element is ignored.

<img width="604" alt="Screenshot 2023-08-15 at 7 38 56 PM" src="https://github.com/mdx-js/eslint-mdx/assets/13206945/07717a29-e53f-4513-9915-20885cd8843f">

### Expected behavior

It should lint both code blocks. I have `mdx/code-blocks` set to true.

### Actual behavior

It only lints the top-level code block

### Runtime

Node v18

### Package manager

npm v9

### OS

macOS

### Build and bundle tools

Other (please specify in steps to reproduce)

**Comment by JounQin** [comment]
code blocks linting is powered by [`eslint-plugin-markdown`](https://github.com/eslint/eslint-plugin-markdown), so maybe you need to post there, but I don't know would they accept this as bug because it's mdx not markdown actually.

cc @btmills

**Comment by btmills** [comment]
From a Markdown syntax perspective, the first set of triple backticks are the text content of an HTML node. Syntax tree [from astexplorer](https://astexplorer.net/#/gist/61270b291582f2234e20c443d957e7e2/2003740f00f2fefe463d8c0fd7fe42042ec892de) (`position` omitted for brevity):

```json
{
  "type": "root",
  "children": [
    {
      "type": "html",
      "value": "<main>\n  ```js\n  im_not_camel_case_oops\n  ```\n</main>"
    },
    {
      "type": "code",
      "lang": "js",
      "meta": null,
      "value": "im_not_camel_case_oops"
    }
  ]
}
```

I'm not familiar with this plugin's implementation, but I'd guess it would need to use an MDX parser instead of a Markdown parser in order to detect code blocks inside MDX nodes.

**Comment by JounQin** [comment]
@btmills Thanks for your reply! If you don't think it should belong to `eslint-plugin-markdown` then I'll take a look how to add this feature in `eslint-plugin-mdx`.

But I still think this is a chance to add this feature in `eslint-plugin-markdown` to support

````md
<main>
  ```js
  im_not_camel_case_oops
  ```
</main>
````

Because it is just a valid markdown?

**Comment by btmills** [comment]
Neither GitHub (nor VSCode, which I have handy) render the `<main>` example snippet you shared as code, which should steer developers away from that style.

Code blocks inside `<details>` should already work without any changes, at least based on the presence of a `code` node in [the AST](https://astexplorer.net/#/gist/6164b6f894ad06e97443cad98b28526a/110ac0049f8af661bbab2c8f47253db95ed307c9):

<details>
  <summary>details demo</summary>

  ```js
  i.am(javascript);
  ```
</details>

**Comment by btmills** [comment]
Oh, hey! [Check this out](https://astexplorer.net/#/gist/6164b6f894ad06e97443cad98b28526a/4d958277469dc4130fb16dbfbf5080ba2bc47748): if you include a blank newline between the `<main>` tag and the opening code block fence, the code block gets parsed and rendered as a code block. Both GitHub and Remark agree on that behavior. So I suppose if you're very strict about how you write MDX, it might be close enough to HTML for the Markdown parser to accept.

````mdx
<main>
  
```js
im_not_camel_case_oops
```
</main>
````

<main>
  
```js
im_not_camel_case_oops
```
</main>

**Comment by JounQin** [comment]
OK, so the problem here is code blocks inside html/jsx works a bit different with plain markdown, right? I'll add this feature in mdx later.

**Comment by JounQin** [comment]
Should be fixed by #564, I'd add a test case for this.
